WorkoutGenerator/GeneratePlan.swift

```diff
@@ -14,111 +14,36 @@ struct GeneratePlan: View {
-                        print("failed")
-                    }
-                }
-                
-            }
-            Button("generate a quad workout") {
-                Task {
-                    do {
-                        //for each enum (one of each of the above muscles) in the pull workout structure, an exercise is added to the exercises array of the pull workout model.
-                        model.quadWorkout.exercises = []
-                        for exercise in model.quadWorkout.structure {
-                            model.quadWorkout.exercises.append(try await model.getExercisesFor(muscle: exercise.rawValue))
-                        }
-                        saveWorkout(workoutType: model.quadWorkout, key: "savedQuadWorkouts")
-                    }
-                    catch {
-                        print("failed")
-                    }
-                }
-                
-            }
-            Button("generate a hamstring workout") {
-                Task {
-                    do {
-                        //for each enum (one of each of the above muscles) in the pull workout structure, an exercise is added to the exercises array of the pull workout model.
-                        model.hamstringWorkout.exercises = []
-                        for exercise in model.hamstringWorkout.structure {
-                            model.hamstringWorkout.exercises.append(try await model.getExercisesFor(muscle: exercise.rawValue))
-                        }
-                        saveWorkout(workoutType: model.hamstringWorkout, key: "savedGluteHamWorkouts")
-                    }
-                    catch {
-                        print("failed")
-                    }
-                }
+        NavigationStack {
+            VStack {
+                Text("Generate a workout")
+                LazyVGrid(columns: [.init(.fixed(100)), .init(.adaptive(minimum: 100))], content: {
+                    ForEach(Model.workoutTypes, id: \.self) { type in
+                        generateWorkoutButtonView(for: type)
+                    }
+                })
             }
+            .frame(width: 250)
         }
     }
     
-    func saveWorkout(workoutType: any Workout, key: String) -> Void {
+    private func generateWorkoutButtonView(for workout: WorkoutType) -> some View {
+        NavigationLink(
+            destination: { PreviewWorkoutView(workoutType: workout) },
+            label: {
+                Text(workout.rawValue)
+                    .frame(width: 100, height: 100)
+                    .foregroundColor(.black)
+                    .background(Color(cgColor: .init(gray: 0.8, alpha: 1)))
+            })
+    }
+    
+    func saveWorkout(workout: Workout, key: String) -> Void {
         
-        if let encoded = try? encoder.encode(workoutType) {
+        if let encoded = try? encoder.encode(workout) {
             defaults.set(encoded, forKey: key)
         } else {
             print("no")
```

Comment:
we need to handle this gracefully hahah I forgot I did that - we could `fatalError()` for now so we know for REAL when it happens lol